parsec/remote_dep_mpi.c

```diff
@@ -1789,6 +1876,37 @@ remote_dep_mpi_put_end_cb(parsec_comm_engine_t *ce,
     return 1;
 }
 
+static int
+remote_dep_mpi_remote_get_end_cb(parsec_comm_engine_t *ce,
+                                 parsec_ce_tag_t tag,
+                                 void *msg,
+                                 size_t msg_size,
+                                 int src,
+                                 void *data)
+{
+    /* Retreive deps from callback_data */
+    remote_dep_cb_data_t *cb_data = (remote_dep_cb_data_t *)msg;
+    parsec_remote_deps_t* deps = cb_data->deps;
+	parsec_execution_stream_t* es = &parsec_comm_es;
+
+    PARSEC_DEBUG_VERBOSE(6, parsec_debug_output, "MPI:\tTO\tna\Get END  \tunknown \tk=%d\twith deps %p\tparams bla\t(tag=bla) data ptr bla",
+            ((remote_dep_cb_data_t *)cb_data)->k, deps);
+
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+
+    TAKE_TIME(es->es_profile, MPI_Data_plds_ek, ((remote_dep_cb_data_t *)cb_data)->k);
+
+    remote_dep_complete_and_cleanup(&deps, 1);
+
+    ce->mem_unregister(cb_data->memory_handle);
+    parsec_thread_mempool_free(parsec_remote_dep_cb_data_mempool->thread_mempools, cb_data);
+
+    parsec_comm_puts--;
```

Comment:
puts or gets ?

Reply:
I don't think that any of `parsec_comm_gets_max`, `parsec_comm_gets`, `parsec_comm_puts_max`, and `parsec_comm_puts` are actually being used in any way, so the point is moot—the number of concurrent communications is being managed by each communication engine, not at the upper layer.